pandas/tests/test_categorical.py

```diff
@@ -519,6 +519,18 @@ def test_contructor_from_categorical_string(self):
         result = Categorical(values, categories=['a', 'b', 'c'], ordered=True)
         tm.assert_categorical_equal(result, expected)
 
+    def test_constructor_with_categorical_categories(self):
+        # GH17884
+        expected = pd.Categorical(['a', 'b'], categories=['a', 'b', 'c'])
+
+        result = pd.Categorical(
+            ['a', 'b'], categories=pd.Categorical(['a', 'b', 'c']))
+        tm.assert_categorical_equal(result, expected)
+
+        result = pd.Categorical(
+            ['a', 'b'], categories=pd.CategoricalIndex(['a', 'b', 'c']))
+        tm.assert_categorical_equal(result, expected)
```

Comment:
`Categorical` and `CategoricalIndex` have been imported, so you can remove the `pd.` here, if you feel that's cleaner.

Reply:
yes, thanks! (updated)